x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/AbstractSearchableSnapshotsTestCase.java

```diff
@@ -190,4 +203,51 @@ protected static void assertThreadPoolNotBusy(ThreadPool threadPool) throws Exce
             }
         }, 30L, TimeUnit.SECONDS);
     }
+
+    /**
+     * Generates one or more cache files using the specified {@link CacheService}. Each cache files have been written at least once.
+     */
+    protected List<CacheFile> randomCacheFiles(CacheService cacheService) throws Exception {
```

Comment:
This has been moved from the PersistentCacheTests class and sightly changed so that it always generate at least one random cache file and always read/write a range in each cache file.